config/peerpods/podvm/azure-podvm-image-handler.sh

```diff
@@ -410,7 +410,64 @@ function add_image_id_annotation_to_peer_pods_cm() {
         "LATEST_IMAGE_ID=${IMAGE_ID}" ||
         error_exit "Failed to add the image id as annotation to peer-pods-cm configmap"
 
-    echo "Image id added as annotation to peer-pods-cm configmap successfully"
+    echo "Image id annotation added as annotation to peer-pods-cm configmap successfully"
+}
+
+# Function to delete the LATEST_IMAGE_ID annotation from the peer-pods-cm configmap
+
+function delete_image_id_annotation_from_peer_pods_cm() {
+    echo "Deleting image id annotation from peer-pods-cm configmap"
+
+    # Check if the peer-pods-cm configmap exists
+    if ! kubectl get configmap peer-pods-cm -n openshift-sandboxed-containers-operator >/dev/null 2>&1; then
+        echo "peer-pods-cm configmap does not exist. Skipping deleting the image id"
+        return
+    fi
+
+    # Delete the image id annotation from peer-pods-cm configmap
+    kubectl annotate configmap peer-pods-cm -n openshift-sandboxed-containers-operator \
+        "LATEST_IMAGE_ID-" ||
+        error_exit "Failed to delete the image id annotation from peer-pods-cm configmap"
+
+    echo "Image id annotation deleted from peer-pods-cm configmap successfully"
+}
+
+# Function to add image gallery annotation to peer-pods-cm configmap
+
+function add_image_gallery_annotation_to_peer_pods_cm() {
+    echo "Adding IMAGE_GALLERY_NAME annotation to peer-pods-cm configmap"
+
+    # Check if the peer-pods-cm configmap exists
+    if ! kubectl get configmap peer-pods-cm -n openshift-sandboxed-containers-operator >/dev/null 2>&1; then
+        echo "peer-pods-cm configmap does not exist. Skipping adding the IMAGE_GALLERY_NAME annotation"
+        return
+    fi
+
+    # Add IMAGE_GALLERY_NAME annotation to peer-pods-cm configmap
+    kubectl annotate configmap peer-pods-cm -n openshift-sandboxed-containers-operator \
```

Comment:
can you explain the rational set the annotations, I think i may have missed something

Reply:
The peer-pods-cm is the single source of truth for dynamically updated info. Hence it's used to store the image as well as the gallery name. The podvm-handling.md explains it as well.